PBlib.py

```diff
@@ -437,9 +446,224 @@ def write_count_matrix(pb_count, outfile, first=1):
         print("%-5d" % (residue_idx + first) +
               " ".join("%5d" % i for i in residue_pb), file=outfile)
 
-#-------------------------------------------------------------------------------
-# vertorize function
-#-------------------------------------------------------------------------------
-angle_modulo_360_vect = numpy.vectorize(angle_modulo_360)
+
+def compute_score_by_position(score_mat, seq1, seq2):
+    """
+    Computes substitution score between two sequences position per position
+
+    The substitution score can represent a similarity or a distance depending
+    on the score matrix provided. The score matrix should be provided as a 2D
+    numpy array with score[i, j] the score to swich the PB at the i-th position
+    in PB.NAMES to the PB at the j-th position in PB.NAMES.
+
+    The function returns the result as a list of substitution scores to go from
+    `seq1` to `seq2` for each position. Both sequences must have the same
+    length.
+
+    ..note:
+
+        The score to move from or to a Z block (dummy block) is always 0.
+
+    Exceptions
+    ----------
+    InvalidBlockError : encountered an unexpected PB
+    """
+    assert len(seq1) == len(seq2), \
+        "sequences have different sizes:\n{}\nvs\n{}".format(seq1, seq2)
+    score = []
+    for pb1, pb2 in zip(seq1, seq2):
+        # score is 0 for Z (dummy PB)
```

Comment:
We should not have to compare "z" to a PB since "z" is not a PB. It just means that a PB cannot be assigned to this residue.
So maybe we should rise an error as this point.

Could you open an issue of this please?